### PR TITLE
bpo-40806: itertools.product docs clarify

### PR DESCRIPTION
**Pull request by stsouko**
Explanation of arguments of product function added.

https://bugs.python.org/issue40806

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@stsouko

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Review by rhettinger — APPROVED** [review]

**Comment by stsouko** [comment]
I don't know why Docs / Docs (pull_request)  test failed

**Comment by rhettinger** [comment]
Standby, I'm working on a fix.

File: Doc/library/itertools.rst [hunk 1]
```diff
@@ -563,6 +563,9 @@ loops that truncate the stream.
            for prod in result:
                yield tuple(prod)
 
+   Before :func:`product` runs, it completely consumes the input iterables,
+   keeping pools of values in memory to generate the products.  Accordingly,
+   it only useful with finite inputs.
```
Comment (mdickinson) [inline Doc/library/itertools.rst:568]: Missing "is" after "it"?

**Pull request merged**